### PR TITLE
Update multicluster installation to match master

### PR DESCRIPTION
**Pull request by sdake**
Big usability improvements have been made.  Document
the new workflow for multicluster.

### CODE REVIEW

File: _docs/setup/kubernetes/multicluster-install.md [hunk 1]
```diff
@@ -52,65 +45,58 @@ has been validated with multicluster.
...
+kubectl label secret ${CLUSTER_NAME} istio/multiCluster=true -n istio-system
+popd
 ```
 
-## Deploy on the remote cluster
+<img src="{{home}}/img/exclamation-mark.svg" alt="Warning" title="Warning" style="width: 32px; display:inline" />Ordering currently matters.  Secrets must be created prior to the deployment of
+the Istio control plane.  Creating secrets after Istio is started will not register the
+secrets with Istio properly.
+
+## Deploy the Istio control plane
```
Comment (gyliu513) [inline _docs/setup/kubernetes/multicluster-install.md]: I'd like we say this as `local` control plane.

File: _docs/setup/kubernetes/multicluster-install.md [hunk 2]
```diff
@@ -52,65 +45,58 @@ has been validated with multicluster.
...
+
+Install the [Istio control plane]({{home}}/docs/setup/kubernetes/quick-start.html)
+on **one** Kubernetes cluster.
 
-Multicluster makes **one** Istio service mesh from all Kubernetes clusters
-that participate in multicluster.  Multicluster uses **one** Kubernetes
-cluster to run the Istio control plane.  **Any reasonable** number of remote
-Kubernetes clusters may be attached to the **one** Istio control plane.
+## Install the Istio remote on every remote
```
Comment (gyliu513) [inline _docs/setup/kubernetes/multicluster-install.md]: s/Install the Istio remote on every remote/Install the Istio remote on every remote cluster

File: _docs/setup/kubernetes/multicluster-install.md [hunk 3]
```diff
@@ -120,7 +106,7 @@ The istio-remote component must be deployed to each remote Kubernetes cluster.
 1. Use the helm template command on a remote to specify the Istio control plane:
 
    ```bash
-   helm template install/kubernetes/helm/istio-remote --name istio-remote --set pilotEndpoint=`pod_ip_of_pilot_master` --set mixerEndpoint=`pod_ip_of_mixer_master` > $HOME/istio-remote.yaml
+   helm template install/kubernetes/helm/istio-remote --name istio-remote --set pilotEndpoint=`pod_ip_of_pilot_local` --set policyEndpoint=`pod_ip_of_policy_local --set statsdEndpoint=`pod_ip_of_statsd_local` > $HOME/istio-remote.yaml
```
Comment (gyliu513) [inline _docs/setup/kubernetes/multicluster-install.md]: What does `pod_ip_of_pilot_local`, `pod_ip_of_policy_local`, `pod_ip_of_statsd_local` means?

Is it mean the pod ip of those components in `local` control plane? Why use pod ip? How can we make sure the pod ip can be accessed cross clusters? How to handle the case if I have multiple pods for pilot?

Comment (gyliu513) [inline _docs/setup/kubernetes/multicluster-install.md]: Losing a "`" for `policyEndpoint`

File: _docs/setup/kubernetes/multicluster-install.md [hunk 4]
```diff
@@ -147,19 +133,20 @@ install one:
 1. Install the Helm chart:
 
    ```bash
-   helm install install/kubernetes/helm/istio-remote --name istio-remote --set pilotEndpoint=pod_ip_of_pilot_master --set mixerEndpoint=pod_ip_of_mixer_master`
+   helm install install/kubernetes/helm/istio-remote --name istio-remote --set pilotEndpoint=`pod_ip_of_pilot_local` --set policyEndpoint=`pod_ip_of_policy_local --set statsdEndpoint=`pod_ip_of_statsd_local` > $HOME/istio-remote.yaml -n istio-system
```
Comment (gyliu513) [inline _docs/setup/kubernetes/multicluster-install.md]: Losing a "`" for `policyEndpoint`

File: _docs/setup/kubernetes/multicluster-install.md (hunk 3 above)
Comment (gyliu513) [reply _docs/setup/kubernetes/multicluster-install.md]: How can I make sure the pod ip in cluster1 can be accessed from cluster2?

Comment (sdake) [reply _docs/setup/kubernetes/multicluster-install.md]: There is no service IP or ILB to use at this time, although that is planned for 0.9.  The pod IP is the only IP address we can currently reference to connect to pilot across clusters.  The requisites cover the requirements that pod networks are routable to one another.  I am not sure how to handle the case of multiple pods for pilot.  Is that something that can be deployed today?

In general I think all of your concerns here will be resolved once ILBs are available in the 0.9 implementation, however, I can't document the future.  Instead only documenting the current state of the system.

File: _docs/setup/kubernetes/multicluster-install.md [hunk 5]
```diff
@@ -29,20 +32,10 @@ pod CIDR.
...
-
-* The local cluster and all remote Kubernetes cluster credentials (typically
-stored in `$HOME/.kube/config`) copied to
-`$HOME/multicluster/${UNIQUE_IDENTIFIER}.kube.conf`.
-
-* A [deployed Istio control plane]({{home}}/docs/setup/kubernetes/quick-start.html)
-on **one** Kubernetes cluster.  Istio must be [deployed with Helm]({{home}}/docs/setup/kubernetes/helm-install.html) using the `--set global.multicluster.enabled=true` flag.  This is currently mandatory.
+* All remote Kubernetes cluster credentials (typically stored in `$HOME/.kube/config`)
+copied to `$HOME/multicluster/${CLUSTER_NAME}`.  All files in `$HOME/multicluster`
+directory must have their cluster name set properly.  Many Kubernetes installers do
```
[the remark this answers is not in the document]
Comment (sdake) [reply _docs/setup/kubernetes/multicluster-install.md]: ${CLUSTER_NAME} in this case is the name of the cluster which the operator specifies for the remotes.  This is highly dependent upon the deployment tooling used.  For example, with kubeadm (which I use for development) - all clusters are named `kubernetes`, requiring a manual rename of the cluster in the credentials file.  In  kubeadm 1.11, the cluster name can be overridden, but I am not keen to document all of the various Kubernetes deployment tool techniques for setting the custer name.

File: _docs/setup/kubernetes/multicluster-install.md [hunk 6]
```diff
@@ -14,9 +14,12 @@ Instructions for the installation of Istio multicluster.
...
 ## Prerequisites
 
-* Two or more Kubernetes clusters.
+* Two or more Kubernetes clusters with **1.7.3 or newer**.
 
-* The usage of an RFC1918 network, VPN, or other more advanced network techniques
+* The ability to deploy the [Istio control plane]({{home}}/docs/setup/kubernetes/quick-start.html)
+on **one** Kubernetes cluster.
+
+* The usage of an RFC1918 network, VPN, or alternative more advanced network techniques
```
Comment (sdake) [inline _docs/setup/kubernetes/multicluster-install.md]: @gyliu513 This is where the pre-reqs are specified requiring pod IPs to be routable between each cluster.

File: _docs/setup/kubernetes/multicluster-install.md (hunk 3 above)
Comment (sdake) [reply _docs/setup/kubernetes/multicluster-install.md]: The technique I use is as follows:

```
sdake@falkor-08:~$ cat route08.sh
#!/bin/bash
ip route add 10.111.0.0/16 via 10.23.220.75
```

And on my other cluster

```
sdake@falkor-07:~$ cat route07.sh
#!/bin/bash
ip route add 10.121.0.0/16 via 10.23.220.76
```

Clearly Kubernetes operators may have different network environments or may be using a VPN, and I don't think it is feasible to document how to setup routing on the pod IPs.

File: _docs/setup/kubernetes/multicluster-install.md [hunk 7]
```diff
@@ -52,65 +45,58 @@ has been validated with multicluster.
...
+plane cluster to `$HOME/multicluster/${CLUSTER_NAME}`.  The
+`${CLUSTER_NAME}` should be unique per remote.
 
-> The implementation only uses the
-`config.istio.io/pilotCfgStore` and `config.istio.io/accessConfigFile`
-annotations, although every other annotation and spec is validated for
-correct syntax.  They may be set to dummy values, as long as they are
-syntactically correct.
+1. Modify the name of the remote cluster's credentials file field
+`clusters.name` to the desired name of the cluster.
```
[the remark this answers is not in the document]
Comment (sdake) [reply _docs/setup/kubernetes/multicluster-install.md]: yes clusters.cluster.name is correct here - nice catch!  desired name of the cluster is equivalent to ${CLUSTER_NAME}.  Any rewording you may suggest that simplifies the understanding would be a valued contribution.

File: _docs/setup/kubernetes/multicluster-install.md (hunk 4 above)
Comment (sdake) [reply _docs/setup/kubernetes/multicluster-install.md]: fixed.

File: _docs/setup/kubernetes/multicluster-install.md (hunk 3 above)
Comment (sdake) [reply _docs/setup/kubernetes/multicluster-install.md]: fixed.

Comment (sdake) [reply _docs/setup/kubernetes/multicluster-install.md]: partially clarified.

File: _docs/setup/kubernetes/multicluster-install.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (sdake) [reply _docs/setup/kubernetes/multicluster-install.md]: clarified.

File: _docs/setup/kubernetes/multicluster-install.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (sdake) [reply _docs/setup/kubernetes/multicluster-install.md]: clarified by removal.  I think this should not be in the prereqs as it was already documented at line 60.

File: _docs/setup/kubernetes/multicluster-install.md (hunk 3 above)
Comment (gyliu513) [reply _docs/setup/kubernetes/multicluster-install.md]: Are you using calico or ovs for your cluster to configure the route?

File: _docs/setup/kubernetes/multicluster-install.md (hunk 6 above)
Comment (gyliu513) [reply _docs/setup/kubernetes/multicluster-install.md]: This is a very big limitation I think, I think that most of the clusters cannot make sure the pod ip is accessible from all clusters.

File: _docs/setup/kubernetes/multicluster-install.md (hunk 3 above)
Comment (gyliu513) [reply _docs/setup/kubernetes/multicluster-install.md]: Do you know how can I enable this if I was using calico?

File: _docs/setup/kubernetes/multicluster-install.md (hunk 6 above)
Comment (gyliu513) [reply _docs/setup/kubernetes/multicluster-install.md]: Thanks @sdake , any github issue opened for this?

File: _docs/setup/kubernetes/multicluster-install.md (hunk 3 above)
Comment (gyliu513) [reply _docs/setup/kubernetes/multicluster-install.md]: @sdake any comments for this? Are you using ovs as an example? Any idea or suggestions if I was using calico for container network? How can I enabled pods communicate cross clusters.

**Pull request merged**